pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -442,17 +449,6 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		return controller.NewPermanentError(err)
 	}
 
-	// Because of parameter propagation, we skip validating it inside the pipelineSpec since it may
-	// not have the full list of defined parameters
-	ctx = config.SkipValidationDueToPropagatedParametersAndWorkspaces(ctx, true)
-	if err := pipelineSpec.Validate(ctx); err != nil {
```

Comment:
+1 - I'm a bit hesitant to drop validation altogether - the definition of what's considered valid may change over time.

A bit curious though that this happens after we build the dag... not sure why this was done. 🤔

Reply:
Going to resolve this comment thread because I think it's the same conversation as in https://github.com/tektoncd/pipeline/pull/6645/files#r1194078732, but feel free to reopen if there's a separate concern.

Reply:
Thanks Lee. I realized on second read that the removal here corresponds to the behavior of validating the referenced Pipeline as if its API version is the storage version, which we would like to change from.